### PR TITLE
[preemption] Two step preemption.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Split the preemption process in two,  target selection and issuing, the first step can be use to check if preemption is possible without issuing preemption to any admitted workloads.
#### Which issue(s) this PR fixes:

Relates to  #420 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
for #805, so that we don't complicate cherry-picks

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -250,6 +255,7 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 			e.inadmissibleMsg = err.Error()
 		} else {
 			e.assignment = flavorassigner.AssignFlavors(log, &e.Info, snap.ResourceFlavors, cq)
+			e.preemptionTargets = s.preemptor.GetTargets(e.Info, e.assignment, &snap)
```
Comment (alculquicondor) [inline pkg/scheduler/scheduler.go:258]: if it's Fit, could we skip `GetTargets`?

Comment (trasc) [reply pkg/scheduler/scheduler.go:258]: sure

**Comment by trasc** [comment]
> /hold
> for #805, so that we don't complicate cherry-picks

There is no conflict or dependency between them, however  I don't have a problem holding it for now.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Comment by alculquicondor** [comment]
> There is no conflict or dependency between them, however I don't have a problem holding it for now

sounds good. But there will be new tests, so that will also be helpful.

**Comment by alculquicondor** [comment]
I'm working on fixing the test infra https://github.com/kubernetes/test-infra/pull/29631

**Comment by trasc** [comment]
/unhold

#805 is now merged

**Comment by alculquicondor** [comment]
Maybe we should remove the lint check from unit tests? It's already done in verify

**Comment by tenzen-y** [comment]
> Yes, can you open that PR?
> 
> Similarly for integration and E2E

Sure.

**Comment by tenzen-y** [comment]
Opened: #823

**Comment by tenzen-y** [comment]
After: https://github.com/kubernetes-sigs/kueue/pull/823

/test pull-kueue-verify-main

**Comment by tenzen-y** [comment]
Oops. we haven't yet fixed `pull-kueue-verify-main` .

**Comment by tenzen-y** [comment]
/hold

Maybe, fixed by #824

**Pull request merged**